src/java/org/apache/cassandra/cql3/functions/AggregateFcts.java

```diff
@@ -836,6 +836,66 @@ public void addInput(Arguments arguments)
         }
     };
 
+    private static abstract class AbstractNativeAggregate implements AggregateFunction.Aggregate
+    {
+        private final AbstractType<?> returnType;
+
+        public AbstractNativeAggregate(AbstractType<?> returnType)
+        {
+            this.returnType = returnType;
+        }
+
+        private ByteBuffer result;
+
+        @Override
+        public void reset()
+        {
+            result = null;
+        }
+
+        @Override
+        public ByteBuffer compute(ProtocolVersion protocolVersion)
+        {
+            return result;
+        }
+
+        @Override
+        public void addInput(Arguments arguments)
+        {
+            ByteBuffer value = arguments.get(0);
+
+            if (value == null)
+                return;
+
+            if (returnType.isReversed())
```

Comment:
can you explain why this is needed vs `returnType.unwrap()`?  If we are comparing we are just reversing the compare, so if we did `returnType.unwrap()` couldn't we avoid all this?